### ISSUE TITLE
Problem with ID... AUTO INCREMENT?

### ISSUE THREAD

**Issue by gassssty** [capacitor-community/sqlite#206]
Hi guys! I'm Gaston

I have a database (server) that works with auto incremental id, so if there are 3 users with id 1,2,3 respectively, if I delete user with id = 3 and then add a new user, the user with id = 4 is added.

But with this plugin, doing the same process, that new user would take the id = 3... Is there any possibility to change this? As this would be a little problem for the server data collection etc....

Thanks (by the way, the plugin works great and is very good!!! :D)

**Comment by jepiqueau** [comment]
@gassssty thanks for using the plugin. I am surprised by this behaviour i will have a look. are you using the iOS or Android part. I will have a look if i can reproduce this

**Comment by gassssty** [comment]
Thanks for replying :D. I'm using Ionic, but it's precisely angular... it's the same in the end but with some changes.

Thankss

**Comment by gassssty** [comment]
So, answering your question, is Android. At the moment Android... but maybe in a future I will use iOS

**Comment by gassssty** [comment]
To help you a little bit, I mean, it's not a problem of the plugin or anything because I know from database that when you put an auto incremental ID, every time you delete an element, that deleted id can be used again... I don't know if you understand me. 

Of course, in detail with an example a little more explained, the problem comes when I do an ''INSERT'' in the plugin DB... the product is inserted with an ID = 20, I make a call to insert in the server database and it is added with ID = 21 because I deleted before the product with ID = 20.
So it makes me to be added in the plugin DB, two products (the same) with ID = 20 and ID = 21... if you want or need more data, tell me.

Thanks anyway :D

**Comment by jepiqueau** [comment]
@gassssty Is your id has been defined with the AUTOINCREMENT argument ie `INTEGER PRIMARY KEY AUTOINCREMENT` SQLite DOES NOT RECOMMEND to use it but in your case in the json object describing the database schema you must have it to reproduce the same behaviout that your server database

```
" The main purpose of using attribute AUTOINCREMENT is to prevent SQLite to reuse a value that has not been used or a value from the previously deleted row.

If you don’t have any requirement like this, you should not use the AUTOINCREMENT attribute in the primary key. "
```

**Comment by jepiqueau** [comment]
@gassssty if you can share your json object it will help but in a mean time i will have to see if i can fix it simply or not you are the first to mention it

**Comment by gassssty** [comment]
Mmm I'm looking at the code and I have the JSON defined like this:

jsonDb: JsonSQLite = {
    "database": "magentoprueba-db",
    "version": 1,
    "encrypted": false,
    "mode": "full",
    "tables": [
      {
        "name": "customer_entity",
        "schema": [
          {
            "column": "id",
            "value": "INTEGER PRIMARY KEY NOT NULL"
          },
          {
            "column": "email",
            "value": "TEXT NOT NULL"
          },
          {
            "column": "firstname",
            "value": "TEXT NOT NULL"
          },
          {
            "column": "lastname",
            "value": "TEXT NOT NULL"
          },
          {
            "column": "password",
            "value": "TEXT"
          },
          {
            "column": "last_modified",
            "value": "INTEGER DEFAULT (strftime('%s', 'now'))"
          }
        ],
        "values": []
      }
    ]
  }

So I am not using the AUTOINCREMENT... I'll test it in a moment and let you know what happens.

Thanks :D

**Comment by gassssty** [comment]
Well, it worked! At the beginning of my project I put the AUTOINCREMENT but I saw that something didn't work and I ended up deleting it for security reasons, in case something went wrong. 
But I have removed the 'NOT NULL', and I have changed it for the AUTOINCREMENT (because never is null) and now it works perfectly. 

It was my fault but anyway, this plugin (hat I've ever try in Angular/Ionic) is the best I've tried because it offers an IMMENSE amount of DB options!!!!!

Thank you very much, if you have any more questions, I know where to contact you :D.

Regards.
